### PR TITLE
[release/v2.20] Allow configuration for front-loadbalancer service of nodeportproxy

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
Signed-off-by: Waleed Malik <ahmedwaleedmalik@gmail.com>

**What does this PR do / Why do we need it**:
Manual cherry-pick of https://github.com/kubermatic/kubermatic/pull/9476

**Does this PR close any issues?**:
Fixes #

**Special notes for your reviewer**:
Reason for manual cherry-pick https://github.com/kubermatic/kubermatic/pull/9476#issuecomment-1084140795

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
- Add support for configuration annotations and loadBalancerSourceRanges for front-loadbalancer service of node port proxy
- For seed CR, spec.NodeportProxy.Annotations is deprecated and spec.NodeportProxy.Envoy.LoadBalancerService.Annotations should be used instead.
```

### CODE REVIEW

File: pkg/apis/kubermatic/v1/datacenter.go [hunk 1]
```diff
@@ -249,7 +267,7 @@ type DatacenterSpec struct {
 
 	//nolint:staticcheck
 	//lint:ignore SA5008 omitgenyaml is used by the example-yaml-generator
-	Fake *DatacenterSpecFake `json:"fake,omitempty,omitgenyaml"`
+	Fake *DatacenterSpecFake `json:"fake,omitempty,omitempty"`
```
Comment (xrstf) [inline pkg/apis/kubermatic/v1/datacenter.go]: Why did you change this?

**Review by xrstf — CHANGES_REQUESTED** [review]
Q

File: pkg/apis/kubermatic/v1/datacenter.go (hunk 1 above)
Comment (ahmedwaleedmalik) [reply pkg/apis/kubermatic/v1/datacenter.go]: I didn't do this explicitly, must have been either some script.

Comment (xrstf) [reply pkg/apis/kubermatic/v1/datacenter.go]: Please undo and re-generate the docs.

Comment (xrstf) [reply pkg/apis/kubermatic/v1/datacenter.go]: This can happen if you run a update-docs.sh script and interrupt it, then the magic combination of `sed` commands will leave you in this half-ready half-broken state.

Comment (ahmedwaleedmalik) [reply pkg/apis/kubermatic/v1/datacenter.go]: Yes, I figured that out. I'm fixing it right now and will fix it in master as well ✌️

Comment (ahmedwaleedmalik) [reply pkg/apis/kubermatic/v1/datacenter.go]: Done

**Review by xrstf — APPROVED** [review]
/approve
